Description: Improves WebSockets performance on high-latency connections
 This resets TCP_NODELAY and increases send buffer on WebSoscket fd
Author: Evgeny Boger <boger@contactless.ru>
Origin: vendor
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: mosquitto-1.4.15/src/websockets.c
===================================================================
--- mosquitto-1.4.15.orig/src/websockets.c	2018-02-28 15:53:44.000000000 +0000
+++ mosquitto-1.4.15/src/websockets.c	2018-03-09 23:41:49.000000000 +0000
@@ -99,7 +99,7 @@
 		"http-only",
 		callback_http,
 		sizeof (struct libws_http_data),
-		0,
+		65535,
 #ifdef LWS_FEATURE_PROTOCOLS_HAS_ID_FIELD
 		0,
 #endif
@@ -112,7 +112,7 @@
 		"mqtt",
 		callback_mqtt,
 		sizeof(struct libws_mqtt_data),
-		0,
+		65535,
 #ifdef LWS_FEATURE_PROTOCOLS_HAS_ID_FIELD
 		1,
 #endif
@@ -125,7 +125,7 @@
 		"mqttv3.1",
 		callback_mqtt,
 		sizeof(struct libws_mqtt_data),
-		0,
+		65535,
 #ifdef LWS_FEATURE_PROTOCOLS_HAS_ID_FIELD
 		1,
 #endif
@@ -226,6 +226,13 @@
 			}
 			mosq->sock = libwebsocket_get_socket_fd(wsi);
 			HASH_ADD(hh_sock, db->contexts_by_sock, sock, sizeof(mosq->sock), mosq);
+
+			int optval = 0;
+			socklen_t optlen = sizeof(optval);
+
+			if (setsockopt(mosq->sock, SOL_TCP, TCP_NODELAY, (const void *)&optval, optlen) < 0)
+				return 1;
+
 			break;
 
 		case LWS_CALLBACK_CLOSED:
